pkg/commands/workload_apply_test.go

```diff
@@ -3279,6 +3280,220 @@ To get status: "tanzu apps workload get my-workload"
+							Git: &cartov1alpha1.GitSource{
+								URL: gitRepo,
+								Ref: cartov1alpha1.GitRef{
+									Branch: gitBranch,
+								},
+							},
+						},
+					},
+				}
+				return append(givenNamespaceDefault, w)
+			}(),
+			ExpectUpdates: []client.Object{
+				&cartov1alpha1.Workload{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: defaultNamespace,
+						Name:      workloadName,
+						Labels: map[string]string{
+							"apps.tanzu.vmware.com/workload-type": "api",
+						},
+					},
+					Spec: cartov1alpha1.WorkloadSpec{
+						Source: &cartov1alpha1.Source{
+							Git: &cartov1alpha1.GitSource{
+								URL: gitRepo,
+								Ref: cartov1alpha1.GitRef{
+									Branch: gitBranch,
+								},
+							},
+						},
+					},
+				},
+			},
+			ExpectOutput: fmt.Sprintf(`
+Update workload:
+...
+  2,  2   |apiVersion: carto.run/v1alpha1
+  3,  3   |kind: Workload
+  4,  4   |metadata:
+  5,  5   |  labels:
+  6     - |    apps.tanzu.vmware.com/workload-type: web
+      6 + |    apps.tanzu.vmware.com/workload-type: api
+  7,  7   |  name: my-workload
+  8,  8   |  namespace: default
+  9,  9   |spec:
+ 10, 10   |  source:
+...
+
+%s
+
+Updated workload %q
+
+To see logs:   "tanzu apps workload tail my-workload"
+To get status: "tanzu apps workload get my-workload"`, clitesting.ToInteractTerminal("? Really update the workload %q? [yN]: y", workloadName), workloadName),
+		},
+		{
+			Name: "update - git source with terminal interaction rejected",
+			Args: []string{workloadName, flags.TypeFlagName, "api"},
+			WithConsoleInteractions: func(t *testing.T, c *expect.Console) {
+				c.ExpectString(clitesting.ToInteractTerminal("? Really update the workload %q? [yN]: ", workloadName))
+				c.Send(clitesting.InteractInputLine("n"))
+				c.ExpectString(clitesting.ToInteractOutput("Skipping workload %q", workloadName))
+			},
+			GivenObjects: func() []client.Object {
+				w := &cartov1alpha1.Workload{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: defaultNamespace,
+						Name:      workloadName,
+						Labels: map[string]string{
+							"apps.tanzu.vmware.com/workload-type": "web",
+						},
+					},
+					Spec: cartov1alpha1.WorkloadSpec{
+						Source: &cartov1alpha1.Source{
+							Git: &cartov1alpha1.GitSource{
+								URL: gitRepo,
+								Ref: cartov1alpha1.GitRef{
+									Branch: gitBranch,
+								},
+							},
+						},
+					},
+				}
+				return append(givenNamespaceDefault, w)
+			}(),
+			ExpectOutput: fmt.Sprintf(`
+Update workload:
+...
+  2,  2   |apiVersion: carto.run/v1alpha1
+  3,  3   |kind: Workload
+  4,  4   |metadata:
+  5,  5   |  labels:
+  6     - |    apps.tanzu.vmware.com/workload-type: web
+      6 + |    apps.tanzu.vmware.com/workload-type: api
+  7,  7   |  name: my-workload
+  8,  8   |  namespace: default
+  9,  9   |spec:
+ 10, 10   |  source:
+...
+
+%s
+
+Skipping workload %q`, clitesting.ToInteractTerminal("? Really update the workload %q? [yN]: ", workloadName), workloadName),
```

Comment:
test case not capturing the response `n` in the expected output

Reply:
Fixed, bad skipping char